### PR TITLE
Book elements implemetation

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/main/java/org/example/bookshop/repository/impl/BookRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.example.bookshop.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@Data
+public class BookRepositoryImpl implements BookRepository {
+
```
Comment (okuzan) [inline src/main/java/org/example/bookshop/repository/impl/BookRepositoryImpl.java]:
```suggestion
```
don't leave empty lines after method/class definition

File: src/main/java/org/example/bookshop/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.example.bookshop.exception.DataProcessingException;
+import org.example.bookshop.model.Book;
+import org.example.bookshop.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@Data
```
Comment (okuzan) [inline src/main/java/org/example/bookshop/repository/impl/BookRepositoryImpl.java]: why do you need Data here?

File: src/main/java/org/example/bookshop/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.example.bookshop.repository.BookRepository;
+import org.example.bookshop.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+
+    @Autowired
+    private BookRepository bookRepository;
```
Comment (okuzan) [inline src/main/java/org/example/bookshop/service/impl/BookServiceImpl.java]:
```suggestion
@Service
@RequiredArgsConstructor
public class BookServiceImpl implements BookService {
    private final BookRepository bookRepository;
```

Comment (okuzan) [reply src/main/java/org/example/bookshop/service/impl/BookServiceImpl.java]: constructor injection is preferred to field injection

File: src/main/java/org/example/bookshop/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+public class Book {
```
Comment (okuzan) [inline src/main/java/org/example/bookshop/model/Book.java]: Don't use @Data on JPA entities, it contains problematic annotations like ToString and EqualsAndHashcode
https://medium.com/@devchaghtai/understanding-and-resolving-lombok-data-pitfalls-in-jpa-hibernate-entities-ebc76000da18
Use @Getter @ Setter

File: src/main/java/org/example/bookshop/BookShopApplication.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        SpringApplication.run(BookShopApplication.class, args);
+    }
+
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            System.out.println("123");
+        };
+    }
+
```
Comment (okuzan) [inline src/main/java/org/example/bookshop/BookShopApplication.java]:
```suggestion
```

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+                <configuration>
+                    <configLocation>${maven.checkstyle.plugin.configLocation}</configLocation>
+                    <consoleOutput>true</consoleOutput>
+                    <failsOnError>true</failsOnError>
+                    <linkXRef>false</linkXRef>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
```
Comment (okuzan) [inline pom.xml]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/bookshop/BookShopApplication.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+@SpringBootApplication
+public class BookShopApplication {
+    public static void main(String[] args) {
+        SpringApplication.run(BookShopApplication.class, args);
+    }
+
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            System.out.println("123");
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/BookShopApplication.java]: implement main according to the task description

File: src/main/java/org/example/bookshop/model/Book.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/Book.java:15]:
```suggestion
@Setter
@Table(name = "books")
```

File: src/main/java/org/example/bookshop/model/Book.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    private String isbn;
+
+    @Column(nullable = false)
+    private BigDecimal price;
+
+    private String description;
+
+    private String coverImage;
+
+    public Book() {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/Book.java]: use Lombok annotation for default constructor

File: src/main/java/org/example/bookshop/repository/impl/BookRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaQuery<Book> criteriaQuery = session.getCriteriaBuilder()
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/repository/impl/BookRepositoryImpl.java]: why do you need here CriteriaBuilder?

File: src/main/java/org/example/bookshop/service/impl/BookServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.RequiredArgsConstructor;
+import org.example.bookshop.model.Book;
+import org.example.bookshop.repository.BookRepository;
+import org.example.bookshop.service.BookService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService {
+    private BookRepository bookRepository;
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/BookServiceImpl.java]:
```suggestion
    private final BookRepository bookRepository;
```

File: src/main/java/org/example/bookshop/model/Book.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    public Book(String title, String author, String isbn, BigDecimal price) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/Book.java]: Where do you use this constructor?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it

File: src/main/java/org/example/bookshop/BookShopApplication.java (hunk 7 above)
Comment (mkazarian-1) [reply src/main/java/org/example/bookshop/BookShopApplication.java]: if I'm not mistake we don't have any requirement about main class in description?

Comment (Elena-Bruyako) [reply src/main/java/org/example/bookshop/BookShopApplication.java]: Checked, you rigth
maybe there are some requirements in video. You need to call methods from your service to check how they work

File: src/main/java/org/example/bookshop/model/Book.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/Book.java]:
```suggestion
```

default constructor will generate automatically

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**